README.md

```diff
@@ -1,85 +1,341 @@
+	
+	![Linking](./images/build-setting.png)
+ 
+
+# Initializing the SDK
+**Note:** Sample projects are included in this repo.
+	 
+- [Swift Sample Project](./sample-apps/swift-sample-app/swift-sample-app.xcworkspace)
+- [ObjC Sample Project](./sample-apps/objc-sample-app/objc-sample-app.xcworkspace)
 
+
+1. ##### Initialize the API with API key.
+	   In your app delegate, on application launch in `application:didFinishLaunchingWithOptions:` method, initialize the Iterable SDK:
+
+	Swift:
+
+	```swift
+	let config = IterableConfig()
+	config.pushIntegrationName = "<your-iterable-push-integration-name>"
+	IterableAPI.initialize(apiKey: "<your-api-key>", launchOptions: launchOptions, config:config)
+	```
+	
+	Objective-C:
+
+	```objective-c
+	IterableConfig *config = [[IterableConfig alloc] init];
+	config.pushIntegrationName = @"<your-iterable-push-integration-name>";
+	[IterableAPI initializeWithApiKey:@"<your-api-key>" launchOptions:launchOptions config:config]
+	```
+	
+	
+	See the Iterable guide on how to setup your Iterable push integration and obtain push integration name [here](https://support.iterable.com/hc/en-us/articles/115000315806-Setting-Up-iOS-Push-Notifications).	
+	
+2. ##### Set userId or email. 
+
+	Once you know the email or userId of the user, set the value.
+
+	Swift:
+	
+	```swift
+	IterableAPI.email = "user@example.com"
 	```
-	IterableAPI.initialize(apiKey: "your-api-key", optionalParameters.....)
+
+	Objective-C:
+
+	```objective-c
+	IterableAPI.email = @"user@example.com";
 	```
-  * Ideally, you will call this from inside `application:didFinishLaunchingWithOptions:` and pass in `launchOptions`. This will let the SDK automatically track a push open for you if the application was launched from a remote Iterable push notification. 
 	
-2. Once you know the email *(Preferred)* or userId of the user, set the value.
+3. ##### Send Remote Notification Token to Iterable
+
+	* See [Apple Notification Guide](https://developer.apple.com/documentation/usernotifications) regarding how to register for remote notifiations.
+	* In your `AppDelegate`’s [application:didRegisterForRemoteNotificationsWithDeviceToken:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622958-application) method, send the token obtained to Iterable.
+
+	Swift:
+	
+	```swift
+	func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        IterableAPI.register(token: deviceToken)
+    }
+    ```
+
+	Objective-C:
+	
+	```objective-c
+	- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+    	[IterableAPI registerToken:deviceToken];
+	}
+    ```
+    See example in sample app delegate [here](./sample-apps/swift-sample-app/swift-sample-app/AppDelegate.swift).
 
+Congratulations! You can now send remote push notifications to your device from Iterable!
```

Comment:
Should we note that this will fail if email or userId was not assigned before this call? (see iOS docs)

Reply:
We do talk about setting userId or email in step 2 but I suppose we can mention as you say and point them to example app here.